src/main/kotlin/com/mparticle/kits/AppboyKit.kt

```diff
@@ -216,6 +244,10 @@ open class AppboyKit : KitIntegration(), AttributeListener, CommerceListener,
 
     override fun logEvent(event: CommerceEvent): List<ReportingMessage> {
         val messages: MutableList<ReportingMessage> = LinkedList()
+        //For CommerceEvent, Event Name is not required to generate hash. So, it will be always null.
+        event.products?.get(0)?.customAttributes?.let {
```

Comment:
you're missing most of what is mappable for a commerce event:
- `event.getTransactionAttributes()`
- `event.customAttributes`
- properties of every product object (eg price, quantity) - you're just doing customAttributes of the 1st product here?

and - does ios or web support mapping properties of impressions and promotions?

Reply:
-  event.getTransactionAttributes() (adding this changes)
-  event.customAttributes(adding this changes)
-  properties of every product object (eg price, quantity) - you're just doing customAttributes of the 1st product here? missed this by mistake, will move to loop for all the products.

Web doesn't support mapping properties of impressions and promotions; and for IOS I think currently it only supports custom events not commerce events.

[https://github.com/mparticle-integrations/mparticle-apple-integration-appboy/blob/[…]cd60edc2751c10129f168d8a/Sources/mParticle-Appboy/MPKitAppboy.m](https://github.com/mparticle-integrations/mparticle-apple-integration-appboy/blob/d741bb617cca5fc9cd60edc2751c10129f168d8a/Sources/mParticle-Appboy/MPKitAppboy.m#L208)